mkosi/__init__.py

```diff
@@ -5689,7 +5690,7 @@ def load_distribution(args: argparse.Namespace) -> argparse.Namespace:
             args.release = r
 
     if args.distribution is None:
-        die("Couldn't detect distribution.")
+        die("Couldn't detect distribution.", ArgsException)
```

Comment:
Ditto. Nothing in parse_args() should throw MkosiNotSupportedException. All errors here should be fatal and not lead to the test being skipped.